lxd/db/networks.go

```diff
@@ -41,6 +42,74 @@ func (c *ClusterTx) GetNetworksLocalConfig(ctx context.Context) (map[string]map[
 	return networks, nil
 }
 
+// GetNetworksNodeParent returns a map associating each given node ID to networks and their
+// node-specific parent value. If network has no parent, it is omitted.
+func (c *ClusterTx) GetNetworksNodeParent(ctx context.Context, nodeID int64, nodeIDs ...int64) (map[int64]map[string]string, error) {
+	var nodeIDsList strings.Builder
+
+	nodeIDs = append(nodeIDs, nodeID)
+
+	first := true
+	for _, id := range nodeIDs {
+		if !first {
+			nodeIDsList.WriteString(",")
+		}
+
+		first = false
+
+		nodeIDsList.WriteString(strconv.FormatInt(id, 10))
+	}
+
+	query := `
+   SELECT networks_config.node_id, networks.name, networks_config.value
+   FROM networks_config
+   JOIN networks ON networks.id=networks_config.network_id
+   WHERE (
+      networks_config.key="parent" AND
+      networks_config.node_id IN (` + nodeIDsList.String() +
```

Comment:
whilst building the query like this is fine in this case (because the input can only be integers), I'd like to see us use query params and `query.Params()` function as a habit, unless theres very specific reasons not to (like in `InstanceList()`).